In [1]:
import numpy as np 

In [3]:
g=np.genfromtxt(r"C:\Users\mdimr\Downloads\Sample_data.csv",delimiter=',',skip_header=1)
g

array([[ 0., 25., 40.],
       [ 1., 30., 35.],
       [ 2., 28., 38.],
       [ 3., 32., 41.],
       [ 4., 26., 37.],
       [ 5., 29., 36.],
       [ 6., 27., 39.],
       [ 7., 33., 42.],
       [ 8., 31., 38.],
       [ 9., 28., 37.],
       [10., 30., 40.],
       [11., 29., 36.]])

# Measures of Central Tendency:

## 1.  Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.


In [5]:
product_A=g[:,1]
product_A

array([25., 30., 28., 32., 26., 29., 27., 33., 31., 28., 30., 29.])

In [6]:
product_B=g[:,2]
product_B

array([40., 35., 38., 41., 37., 36., 39., 42., 38., 37., 40., 36.])

In [8]:
x=np.mean(product_A)
x

29.0

In [9]:
y=np.mean(product_B)
y

38.25

In [29]:
print(x,y)

29.0 38.0


*  Average monthly sales of product A  is 29.0 and product B  is 38.0
    * 1. sales of product A is lower than product B 
    * 2. The sales of product B is higher over months  
    * Hence product B perform better sales than product A

## 2. Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.


In [15]:
x=np.median(product_A)
x

29.0

*  median of product A is middle value it indicates that below median sales are low and above medium sales are high

In [97]:
y=np.median(product_B)
y

38.0

*  median of product B is middle value it indicates that below median sales are low and above medium sales are high

* median shows a clear middle point in the sales data for both products 

## 3. Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.
#### Mode of product A

In [23]:
import statistics

In [24]:
statistics.mode(product_A)

30.0

#### Mode of product B

In [25]:
statistics.mode(product_B)

40.0

* The mode represents the most frequent sales value in the dataset. 
* For Product A, having a mode of 30 means that 30 is the sales figure that occurs most often. 
* Similarly, for Product B, a mode of 40 indicates that 40 is the most common sales value.
* Products with a mode close to the mean and median suggest a relatively symmetrical distribution
* differences between the mode and other measures may indicate skewness.

## 4. Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.

##### weighted mean is each month has an equal weight 

In [36]:
len(product_A)

12

In [40]:
weight=np.ones(12)
weight

array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])

### weighted mean of product A

In [45]:
weighted_A=weight * product_A
weighted_A

array([25., 30., 28., 32., 26., 29., 27., 33., 31., 28., 30., 29.])

In [47]:
np.mean(weighted_A)

29.0

### weighted mean of product B

In [46]:
weighted_B=weight * product_B
weighted_B

array([40., 35., 38., 41., 37., 36., 39., 42., 38., 37., 40., 36.])

In [48]:
np.mean(weighted_B)

38.25

# Measures of Dispersion:
## 1. Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.
#### range of product A

In [49]:
product_A

array([25., 30., 28., 32., 26., 29., 27., 33., 31., 28., 30., 29.])

In [66]:
range_A=np.max(product_A)-np.min(product_A)
range_A

8.0

* range_A suggests that monthly sales varied by upto 8 units over observed period 

#### range of product B

In [62]:
product_B

array([40., 35., 38., 41., 37., 36., 39., 42., 38., 37., 40., 36.])

In [65]:
range_B=np.max(product_B)-np.min(product_B)
range_B

7.0

* range B suggests that monthly sales varied by upto 7 units over observed period 

* product B has larger range representing more variability in monthly sales compared to product A

## 2. Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.
####  variance for  product A

In [71]:
var_A=np.var(product_A)
var_A

5.166666666666667

####  variance for product B

In [72]:
var_B=np.var(product_B)
var_B

4.354166666666667

* A higher variance indicates that the sales figures for Product A are more spread out from the mean compared to Product B
* A lower variance, as seen in Product B, implies that its sales data points are closer to the mean. This suggests that Product B has a more consistent and stable sales performance compared to Product A. 

## 3. Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.
#### standard deviation for Product A

In [75]:
StdVar_A=np.std(product_A)
StdVar_A

2.273030282830976

#### standard deviation for Product B

In [77]:
StdVar_B=np.std(product_B)
StdVar_B

2.0866640042581523

* standard deviation is a measure of the amount of variation or dispersion in a set of values 
* it is the squareroot of  variance 
* product  B is with lower standard deviation , this indicates more stable and consistent sales performence with less variabilty
* product A is with Higher standard deviation suggests greater variability in monthly sales, this indicates sales performence with higher variabilty unstable and inconsistent of sales 

# Relative Measures of Dispersion:
## 1. Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.
#### Coefficient of Variation of product A

* The coefficient of variation is a relative measure of dispersion that expresses the standard deviation as a percentage of the mean.
* higher coefficient of variation indicates a greater relative variability in sales compared to the mean.

In [78]:
y=np.mean(product_A)
StdVar_A=np.std(product_A)
coeff_Var_A=(StdVar_A/x)*100
coeff_Var_A

7.838035458037848

* product A mean is 29.0  and product A coefficient of variation is 7.838035458037848 
* A product with a low CV has more stable sales, while a product with a high CV has more fluctuating sales.
* A lower CV indicates a less relative variability compared to the mean,  hence product A has more stable sale values 

## 2. Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.
#### Relative Range  of product B

In [94]:
range_B=np.max(product_B)-np.min(product_B)
y=np.mean(product_B)
rel_range_B=(range_B/y)*100
rel_range_B

18.30065359477124

* product B is mean is 38.25 and relative range of product B = 18.30065359477124 
* A product with a high mean sales but a low relative range has more stable sales than a product with a low mean sales but a high relative range.
* The relative range of product B  can help in  comparing  the variability of products with different scales of sales.
* Here  product-B is with a high mean sales but a low relative range  so, product B has more stable sales

## 3. Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.
#### Relative standard deviation for product A

In [96]:
x=np.mean(product_A)
StdVar_A=np.std(product_A)
rel_std_A=(StdVar_A/x)*100
rel_std_A

7.838035458037848

* Product A mean is 29.0 and relative standard deviation of product A is  7.838035458037848
* Product A is  with low CV values have less relative variability around their mean, sales are stable consistent 